assets/model_monitoring/components/tests/e2e/conftest.py

```diff
@@ -609,3 +610,45 @@ def uploaded_version_file_name():
 def components_are_uploaded(uploaded_version_file_name):
     """Check if we already have a component version available uploaded by github CI."""
     return os.path.exists(uploaded_version_file_name)
+
+
+@pytest.fixture(scope="module")
+def submit_pipeline_job(ml_client: MLClient, request):
+    """Submit pipeline job to ml ws. Handle deletion if the testcase is cancelled/stopped."""
+    submitted_jobs = []
+
+    def _submit_job(job: Job, experiment_name: str):
+        pipeline_job = ml_client.jobs.create_or_update(
+            job, experiment_name=experiment_name, skip_validation=True
+        )
+        submitted_jobs.append(pipeline_job)
+        return pipeline_job
+
+    def cancel_jobs():
+        print("cancelling the jobs submitted:")
+        for job in submitted_jobs:
+            job_at_end = ml_client.jobs.get(job.name)
+            print(f"job: {job.name}, status: {job.status}, end_status: {job_at_end.status}")
+            if job_at_end.status not in RunHistoryConstants.TERMINAL_STATUSES:
+                ml_client.jobs.begin_cancel(job.name)
+
+    request.addfinalizer(cancel_jobs)
+    yield _submit_job
+
+
+@pytest.fixture(scope="session")
+def cleanup_previous_e2e_tests(ml_client: MLClient, test_suite_name):
+    """Clean up any previously running e2e tests in test suite name."""
+    print(f"Cleaning up past jobs that weren't cancelled in experiment_name={test_suite_name}.")
+    if test_suite_name == "local":
```

Comment:
we might need to notify team that do not override the experiment name for e2e test, or need to find another way to check the local job vs ci job. For example, submitter identity or maybe add an option parameter is_local which can pass to this function

Reply:
Yes that's true. I didn't see any of the e2e tests overriding the experiment name but it would be a good idea to mention that for developers. I am not sure how much more work it would be to idenitfy the submitter identity vs just using the test_suite_name fixture

Reply:
In a way test_suite_name fixture is already the optional parameter since it looks in the environment variables for "GITHUB_REF_NAME" and majority of people don't set that so that is how we know it is local